File: 75_Sort_Colors.c
void sortColors(int* nums, int numsSize) {
    int a[] = {0, 0, 0};

    for(int i= 0; i < numsSize; i++){
        if(nums[i] == 0){
            a[0]++;
        }
        else if(nums[i] == 1){
            a[1]++;
        }
        else{
            a[2]++;
        }
    }

    int index = 0;
    for(int i = 0; i < a[0]; i++){
        nums[index++] = 0;
    }
    for(int i = 0; i < a[1]; i++){
        nums[index++] = 1;
    }
    for(int i = 0; i < a[2]; i++){
        nums[index++] = 2;
    }
}

int main() {
    int nums[] = {2, 0, 2, 1, 1, 0};
    int numsSize = sizeof(nums) / sizeof(nums[0]);

    sortColors(nums, numsSize);

    for(int i = 0; i < numsSize; i++) {
        printf("%d ", nums[i]);
    }
    return 0;
}

File: 75_Sort_Colors.py
class Solution:
    def sortColors(self, nums: list[int]) -> None:
        '''        This is the Dutch National Flag algorithm.        
        low = mid = 0
        high = len(nums) - 1
        
        while(mid <= high):
            if nums[mid] == 0:
                nums[low], nums[mid] = nums[mid], nums[low]
                low += 1
                mid += 1
            elif nums[mid] == 1:
                mid += 1
            elif nums[mid] == 2:
                nums[high], nums[mid] = nums[mid], nums[high]
                high -= 1
        '''
        d = {}
        for i in nums:    
            d[i] = d.get(i,0) + 1

        nums.clear()
        for i in range(3):
            if i in d:
                nums.extend([i] * d[i])



a = Solution()
x = [2,0,2,1,1,0]
a.sortColors(x)
print(x)